src/App.tsx

```diff
@@ -1,33 +1,16 @@
 import { useEffect, useRef, useState } from 'react';
 import { DefaultEventsMap } from '@socket.io/component-emitter';
 import Player from './Player';
-import './App.css';
 import { Socket, io } from 'socket.io-client';
+import './App.css';
 
 let socket: Socket<DefaultEventsMap, DefaultEventsMap>;
 
 function App() {
   const borderRef = useRef<HTMLDivElement>(null);
   const [clients, setClients] = useState<string[]>([]);
-  const [num, setNum] = useState(0);
-  console.log('clients!!!', clients);
-
-  const addClient = (clientId: string) => {
-    if (!clients.includes(clientId)) {
-      console.log('playerConnect', clientId, clients, num);
-      console.log(clients.concat([clientId]));
-      setClients(clients.concat([clientId]));
```

Comment:
It's annoying to set array state in React. Something about immutability. I like to think of it sometimes like a linked list, where you need to have easy access to the "previous" value of the state, which will allow to you know what to update, and then achieve your desired "current" value.

instead of:
```typescript
setClients(clients.concat([clientId]));
```

you would want to do:
```typescript
setClients((previousClients) => [...previousClients, clientId])
```
(`clientId` being the newly connected client).

But since we now can just maintain a list of clients in the server, we don't have to deal with FE state :)

Reply:
Hope that's helpful!